SlevomatCodingStandard/Sniffs/Classes/PropertyPerClassLimitSniff.php

```diff
@@ -0,0 +1,54 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\Classes;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SlevomatCodingStandard\Helpers\ClassHelper;
+use SlevomatCodingStandard\Helpers\SniffSettingsHelper;
+use function sprintf;
+use const T_ANON_CLASS;
+use const T_CLASS;
+use const T_TRAIT;
+
+final class PropertyPerClassLimitSniff implements Sniff
+{
+
+	public const CODE_PROPERTY_PER_CLASS_LIMIT = 'PropertyPerClassLimit';
+
+	/** @var int */
+	public $maxPropertyCount = 10;
+
+	/**
+	 * @return array<int, (int|string)>
+	 */
+	public function register(): array
+	{
+		return [T_CLASS, T_TRAIT, T_ANON_CLASS];
+	}
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+	 * @param File $file
+	 * @param int $classPointer
+	 */
+	public function process(File $file, $classPointer): void
+	{
+		$propertiesCount = ClassHelper::getPropertiesCount($file, $classPointer);
+
+		if ($propertiesCount <= SniffSettingsHelper::normalizeInteger($this->maxPropertyCount)) {
+			return;
+		}
+
+		$tokenType = $file->getTokens()[$classPointer]['content'];
+		$errorMessage = sprintf(
+			'"%s" has too many properties: %d. Can be up to %d properties.',
+			$tokenType,
+			$propertiesCount,
+			$this->maxPropertyCount
+		);
```

Comment:
Hi @50bhan, I don't know if this is an actual issue or not, but the original value of `$this->maxPropertyCount` is used in this `sprintf` command, but is converted to an integer using the `%d` specifier. This conversion may or may not be different from that of `SniffSettingsHelper::normalizeInteger`.

In the error printing of sniff properties in most other sniffs, the normalized value is passed to the`sprintf`.